From 15330cf91b3853540ed3f891f801a4639c9d21ef Mon Sep 17 00:00:00 2001
From: "MeiChia.Chiu" <MeiChia.Chiu@mediatek.com>
Date: Mon, 3 Jun 2024 14:29:24 +0800
Subject: [PATCH 091/236] mtk: hostapd: Fix the issue with the presence of
 MLD_ID in the probe response

The probe response carries the MLD_ID value only
when mld_id is not 0 (e.g., NonTxBSS).

Signed-off-by: Evelyn Tsai <evelyn.tsai@mediatek.com>
Signed-off-by: Money Wang <money.wang@mediatek.com>
Signed-off-by: MeiChia Chiu <MeiChia.Chiu@mediatek.com>
---
 src/ap/ieee802_11_eht.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/ap/ieee802_11_eht.c b/src/ap/ieee802_11_eht.c
index ee3ea46b4..e82bc2e6c 100644
--- a/src/ap/ieee802_11_eht.c
+++ b/src/ap/ieee802_11_eht.c
@@ -497,7 +497,7 @@ u8 * hostapd_eid_eht_basic_ml_common(struct hostapd_data *hapd,
 	if (hapd->iconf->eml_disable)
 		common_info_len -= 2; /* EML Capabilities (2) */
 
-	if (include_mld_id) {
+	if (include_mld_id && hostapd_get_mld_id(hapd)) {
 		/* AP MLD ID */
 		control |= BASIC_MULTI_LINK_CTRL_PRES_AP_MLD_ID;
 		common_info_len++;
@@ -545,7 +545,7 @@ u8 * hostapd_eid_eht_basic_ml_common(struct hostapd_data *hapd,
 		   mld_cap);
 	wpabuf_put_le16(buf, mld_cap);
 
-	if (include_mld_id) {
+	if (include_mld_id && hostapd_get_mld_id(hapd)) {
 		wpa_printf(MSG_DEBUG, "MLD: AP MLD ID=0x%x",
 			   hostapd_get_mld_id(hapd));
 		wpabuf_put_u8(buf, hostapd_get_mld_id(hapd));
@@ -915,8 +915,9 @@ size_t hostapd_eid_eht_ml_beacon_len(struct hostapd_data *hapd,
 				     struct mld_info *info,
 				     bool include_mld_id)
 {
-	return hostapd_eid_eht_ml_len(info, include_mld_id, false,
-				      hapd->iconf->eml_disable);
+	return hostapd_eid_eht_ml_len(info,
+				      include_mld_id && hostapd_get_mld_id(hapd),
+				      false, hapd->iconf->eml_disable);
 }
 
 
-- 
2.45.2

